.github/workflows/bare.yaml

```
@@ -60,9 +52,8 @@ jobs:
     - name: Upload testing package
       run: bash -c './publish_dist.sh ; rm priv_key'
 
-    # Command line debugging through Tmate. v3 works on Windows too.
     - name: Tmate debug on failure
-      if: failure() && env.TMATE_DEBUG == 'yes'
+      if: ${{ failure() && !cancelled() && runner.debug }}
```

One thing that I did, is trigger tmage only for manual `debug` re-runs.

You can trigger a re-run with debug enabled only for a single job

And another idea is to move the `tmate` step before the test run
ON a manual re-run you can manually re-run the failed test, and don't have to wait that long before tmate is available
  
```
  - name: Test Pythia
      run: bash ./build.sh test
```

```suggestion
      if: ${{ !cancelled() && runner.debug }}
```

---

Committed suggestion through 22ea68f5.